qiskit/aqua/algorithms/vq_algorithm.py

```diff
@@ -98,9 +98,6 @@ def var_form(self, var_form: Union[QuantumCircuit, VariationalForm]):
                 "Unsupported type '{}' of var_form".format(
                     type(var_form) if var_form else var_form))
 
-        if var_form is not None and len(self._var_form_params) == 0:
```

Comment:
Why is this check in the base class being removed? For a variational algo don't we want to assure the variational cct has some parameters that can be varied?

Reply:
I think I had this also in the `_energy_evaluation` so it was a duplicate check. But it makes more sense to check this in the setter, I'll move it there.

Reply:
Edit: This needs to go the energy_evaluation or construct_circuit because we can pass `var_form=RY()` which doesn't have a defined num params.